core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -937,6 +950,22 @@ public static Expression translateLiteral(
     return list;
   }
 
+  private List<Expression> translateTableFunctionCall(RexCall rexCall) {
+    if (rexCall.op.getKind() == SqlKind.TUMBLE) {
+      assert rexCall.getOperands().get(2) instanceof RexLiteral;
+      Expression intervalExpression = translate(rexCall.getOperands().get(2));
```

Comment:
We may firstly assert that the operands number is correct to avoid `INDEX OUT OF BOUNDES`. We may also give some messages to these assertions.

Reply:
Will do.

Reply:
addressed